pkg/network/multus/status_test.go

```diff
@@ -33,72 +32,117 @@ import (
+			}
+
+			result := multus.NonDefaultNetworkStatusIndexedByIfaceName(newStubPod(annotations))
+
+			expectedResult := map[string]networkv1.NetworkStatus{
+				"net1": {
+					Name:      "meganet",
+					Interface: "net1",
+					Mac:       "8a:37:d9:e7:0f:18",
+					Default:   false,
+				},
+			}
+
+			Expect(result).To(Equal(expectedResult))
+		})
 	})
 
-	It("Should return a map of pod interface name to network status when interface name is ordinal", func() {
-		const (
-			multusOrdinalNetworksAnnotation                       = `[{"name":"meganet","namespace":"default","interface":"net1"}]`
-			multusNetworkStatusWithPrimaryAndOrdinalSecondaryNets = `[` +
-				`{"name":"k8s-pod-network","ips":["10.244.196.146","fd10:244::c491"],"default":true,"dns":{}},` +
-				`{"name":"meganet","interface":"net1","mac":"8a:37:d9:e7:0f:18","dns":{}}` +
-				`]`
+	Context("NetworkStatusesFromPod", func() {
+		DescribeTable("should return an empty slice", func(podAnnotations map[string]string) {
+			Expect(multus.NetworkStatusesFromPod(newStubPod(podAnnotations))).To(BeEmpty())
+		},
+			Entry("when network status annotation is missing",
+				map[string]string{},
+			),
+			Entry("when network status annotation is empty",
+				map[string]string{networkv1.NetworkStatusAnnot: ""},
+			),
+			Entry("when network status annotation is illegal",
+				map[string]string{networkv1.NetworkStatusAnnot: "not a valid JSON array"},
+			),
 		)
 
-		annotations := map[string]string{
-			networkv1.NetworkAttachmentAnnot: multusOrdinalNetworksAnnotation,
-			networkv1.NetworkStatusAnnot:     multusNetworkStatusWithPrimaryAndOrdinalSecondaryNets,
-		}
-
-		result := multus.NonDefaultNetworkStatusIndexedByIfaceName(newStubPod(annotations))
-
-		expectedResult := map[string]networkv1.NetworkStatus{
-			"net1": {
-				Name:      "meganet",
-				Interface: "net1",
-				Mac:       "8a:37:d9:e7:0f:18",
-				Default:   false,
-			},
-		}
+		It("Should return a valid network status slice", func() {
+			const multusNetworkStatusWithPrimaryAndSecondaryNets = `[` +
+				`{"name":"k8s-pod-network","ips":["10.244.196.146","fd10:244::c491"],"default":true,"dns":{}},` +
+				`{"name":"meganet","interface":"pod7e0055a6880","mac":"8a:37:d9:e7:0f:18","dns":{}}` +
+				`]`
 
-		Expect(result).To(Equal(expectedResult))
+			annotations := map[string]string{networkv1.NetworkStatusAnnot: multusNetworkStatusWithPrimaryAndSecondaryNets}
+
+			expectedResult := []networkv1.NetworkStatus{
+				{
+					Name:    "k8s-pod-network",
+					IPs:     []string{"10.244.196.146", "fd10:244::c491"},
+					Default: true,
+					DNS:     networkv1.DNS{},
+				},
+				{
+					Name:      "meganet",
+					Interface: "pod7e0055a6880",
+					Mac:       "8a:37:d9:e7:0f:18",
+					Default:   false,
+					DNS:       networkv1.DNS{},
+				},
+			}
+
+			Expect(multus.NetworkStatusesFromPod(newStubPod(annotations))).To(Equal(expectedResult))
```

Comment:
> I don't insist on this but I generally think that it is always better practice to use [ConsistOf](https://onsi.github.io/gomega/#consistofelement-interface) when comparing multi item lists. The logic here may change in the future and order may not remain guaranteed,

A list/slice has order, so it is unclear to me why a unit test cannot assure it is maintained and preserved.
I agree it is not always a fit, but it depends on the specification/API of the expectation.

 > furthermore Gomega's message may is better tailored for slices in case of errors
 
For that you could suggest https://onsi.github.io/gomega/#haveexactelementselement-interface.
On which there will be probably no dispute.